src/libs/actions/TeachersUnite.js

```diff
@@ -0,0 +1,174 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import Navigation from '../Navigation/Navigation';
+import CONST from '../../CONST';
+import * as ReportUtils from '../ReportUtils';
+import * as OptionsListUtils from '../OptionsListUtils';
+
+let sessionEmail = '';
+let sessionAccountID = 0;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (val) => {
+        sessionEmail = lodashGet(val, 'email', '');
+        sessionAccountID = lodashGet(val, 'accountID', 0);
+    },
+});
+
+let allPersonalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (val) => (allPersonalDetails = val),
+});
+
+/**
+ * @param {String} firstName
+ * @param {String} phoneOrEmail
+ * @param {String} [lastName]
+ */
+function referTeachersUniteVolunteer(firstName, phoneOrEmail, lastName = '') {
+    const reportID = CONST.TEACHER_UNITE.PUBLIC_ROOM_ID;
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                firstName,
+                lastName,
+                phoneOrEmail,
+            },
+        },
+    ];
+    API.write(
+        'ReferTeachersUniteVolunteer',
+        {
+            reportID,
+            firstName,
+            lastName,
+            phoneOrEmail,
+        },
+        {optimisticData},
+    );
+    Navigation.dismissModal(CONST.TEACHER_UNITE.PUBLIC_ROOM_ID);
+}
+
+/**
+ * Optimistically creates a policyExpenseChat for schoolPrincipal and pass data to AddSchoolPrincipal
+ * @param {String} firstName
+ * @param {String} email
+ * @param {String} [lastName]
+ */
+function createExpenseChatSchoolPrincipal(firstName, email, lastName = '') {
```

Comment:
Let's use the same name of the command for this action. The description already mentions that we're creating a policyExpenseChat for the schoolPrincipal

```suggestion
function addSchoolPrincipal(firstName, email, lastName = '') {
```